server/src/main/java/org/opensearch/index/engine/InternalEngine.java

```diff
@@ -293,7 +295,8 @@ public void onFailure(String reason, Exception ex) {
                     translogManager::getLastSyncedGlobalCheckpoint
                 );
                 this.localCheckpointTracker = createLocalCheckpointTracker(localCheckpointTrackerSupplier);
-                writer = createWriter();
+                this.indexWriterConfig = getIndexWriterConfig();
+                writer = createWriter(indexWriterConfig);
```

Comment:
Just to confirm -- this change will only be "semi-dynamic". 

That is, you need to restart the `Engine` to recreate the `IndexWriterConfig` and recreate the `IndexWriter`, right? Is that intentional?

If you call the `getIndexWriterConfig()` method on `InternalEngine`, you get a copy of the `LiveIndexWriterConfig` which affects the current `IndexWriter`.

I think it might be better to call `getIndexWriterConfig()` from the `onSettingsChanged` method to update the `LiveIndexWriterConfig`. What do you think, @kkewwei?

Reply:
By the change, we can dynamic change the setting without restart the Engine.